src/file_resolver.rs

```diff
@@ -40,25 +40,14 @@ pub struct ResolvedFile {
 impl FileResolver {
     /// Add import path
     pub fn add_import_path(&mut self, path: &Path) -> io::Result<()> {
-        self.import_paths.push((None, path.canonicalize()?));
+        self.import_paths.push((None, path.to_path_buf()));
         Ok(())
     }
 
     /// Add import map
     pub fn add_import_map(&mut self, map: OsString, path: PathBuf) -> io::Result<()> {
-        if self
-            .import_paths
-            .iter()
-            .any(|(m, _)| m.as_ref() == Some(&map))
-        {
-            Err(Error::new(
-                ErrorKind::Other,
-                format!("duplicate mapping for '{}'", map.to_string_lossy()),
-            ))
-        } else {
-            self.import_paths.push((Some(map), path.canonicalize()?));
-            Ok(())
-        }
+        self.import_paths.push((Some(map), path));
```

Comment:
solc allows duplicate mappings and just uses the last one. I think this is desirable and correct (e.g., this is easier for debugging large generated compilation configs)

Reply:
I don't follow why allowing duplicate mappings is desirable. The entries before the last one get silently dropped.

How about warning that a mapping is being overwritten by another one by the same name?

Reply:
Yeah, I'm happy to add a warning. I definitely think it's better to succeed than to fail (I'll again appeal to the build system argument from above).